client/db/src/lib.rs

```diff
@@ -957,6 +958,11 @@ impl<Block: BlockT> Backend<Block> {
 		// TODO: ensure best chain contains this block.
 		let number = *header.number();
 		self.ensure_sequential_finalization(header, last_finalized)?;
+
+		if self.finalize_blacklist.read().contains(hash) {
+			return Ok((*hash, number, false, false))
```

Comment:
Why not return an error?

Reply:
In this case although we don't finalize the block, we still want to keep importing the block, right?
If we return error here, I think the overall importing would fail.